### PR TITLE
Replace deprecated WINAPI GetVersionInfoEx

### PR DESCRIPTION
**Pull request by adrido**
The winapi [GetVersionInfoEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451(v=vs.85).aspx) is deprecated since Windows 8.1 SDK and does only return Windows 8 (6.2) on newer Versions (And prints a warning). One solution would be to manifest the minetest.exe as compatible with the newer Windows versions, but since MinGW builds are not manifested, this would not work there. 
Microsoft recommends to either read the version from kernel32.dll or read it from the Registry.
Since Minetest did never use the Registry, this PR reads the version from kernel32.dll 

[GetVersionInfoEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451(v=vs.85).aspx) might never removed in any future windows versions due to compatibility reasons, but it might removed in future SDK's

### CODE REVIEW

File: src/porting.cpp [hunk 1]
```diff
@@ -173,28 +173,38 @@ bool detectMSVCBuildDir(const std::string &path)
 std::string get_sysinfo()
 {
 #ifdef _WIN32
-	OSVERSIONINFO osvi;
+	#pragma comment(lib, "version.lib")
```
Comment (sfan5) [inline src/porting.cpp]: causes a warning: https://travis-ci.org/minetest/minetest/jobs/283642699#L4876

File: src/porting.cpp [hunk 2]
```diff
@@ -173,28 +173,38 @@ bool detectMSVCBuildDir(const std::string &path)
...
+	delete[] systemDir;
+
+	oss << "Windows/"
+		<< HIWORD(fixedFileInfo->dwProductVersionMS) << '.' // Major
+		<< LOWORD(fixedFileInfo->dwProductVersionMS) << '.' // Minor
+		<< HIWORD(fixedFileInfo->dwProductVersionLS) << ' '; // Build
+
 	#ifdef _WIN64
-	oss << "x86_64";
+		oss << "x86_64";
```
Comment (sfan5) [inline src/porting.cpp]: uh?

File: src/porting.cpp [hunk 3]
```diff
@@ -173,28 +173,38 @@ bool detectMSVCBuildDir(const std::string &path)
...
-	if (osvi.szCSDVersion[0])
-		oss << "-" << tmp;
-	oss << " ";
+	char *systemDir = new char[MAX_PATH];
+	UINT blockSize;
+	VS_FIXEDFILEINFO *fixedFileInfo;
+
+	GetSystemDirectoryA(systemDir, MAX_PATH);
+
+	LPCSTR filePath = std::string(systemDir).append("\\kernel32.dll").c_str();
```
Comment (sfan5) [inline src/porting.cpp]: surely `PathAppendA(systemDir, "kernel32.dll);` is easier

File: src/porting.cpp (hunk 2 above)
Comment (adrido) [reply src/porting.cpp]: "uh?" because its x86_64 instead of x64 or "uh?" because of the corrected indent?

File: src/porting.cpp (hunk 3 above)
Comment (adrido) [reply src/porting.cpp]: thanks

File: src/porting.cpp (hunk 2 above)
Comment (sfan5) [reply src/porting.cpp]: because indentation

Comment (adrido) [reply src/porting.cpp]: The sense of better readability? But ok I can live with that.

**Comment by adrido** [comment]
Thanks.

**Pull request merged**